packager/packager.cc

```diff
@@ -1282,9 +1314,40 @@ ast::ParsedFile validatePackage(core::Context ctx, ast::ParsedFile file) {
     if (!absPkg.exists()) {
         // We already produced an error on this package when producing its package info.
         // The correct course of action is to abort the transform.
+
         return file;
     }
 
+    auto &pkgInfo = PackageInfoImpl::from(absPkg);
+    for (auto &i : pkgInfo.importedPackageNames) {
+        auto &otherPkg = packageDB.getPackageInfo(i.name.mangledName);
+
+        // this might mean the other package doesn't exist, but that
+        // should have been caught already
+        if (!otherPkg.exists()) {
+            continue;
+        }
+
+        if (!otherPkg.visibleTo().empty()) {
```

Comment:
Maybe just:

```suggestion
        const auto &visibleTo = otherPkg.visibleTo();
        if (visibleTo.empty()) {
            continue;
        }
```

And then you can a) dedent the code below and b) reuse `visibleTo`.